logstash-rs/src/buffer.rs

```diff
@@ -39,18 +41,29 @@ impl BufferedSender {
 
 impl Sender for BufferedSender {
     fn send(&self, event: LogStashRecord) -> Result<()> {
-        self.sender.send(Command::Send(event)).map_err(From::from)
+        let result = self.sender.try_send(Command::Send(event));
+        process_result(result, event.level <= Level::Warn)
     }
 
     fn send_batch(&self, events: Vec<LogStashRecord>) -> Result<()> {
-        self.sender.send(Command::SendBatch(events))?;
-        Ok(())
+        let important = events.iter().any(|e| e.level <= Level::Warn);
```

Comment:
This gives an overhead event when there is no error. You should estimate importance only if error happened (after `try_send`)

Reply:
But events is being moved into command, so it can only be checked before